### ISSUE TITLE
Dota crashes during runtime after 7.33 update. Error CPackedStore::TestHashChunkFraction(): Content file pak01.vpk is corrupt

### ISSUE THREAD

**Issue by 1nuyasha** [ValveSoftware/Dota2-Gameplay#8584]
### Description

 I occasionally experience game crashes while playing Dota 2 after 7.33 update (about 10 times in total in these days), and an error message pops up asking me to verify the integrity of game files. However, after verifying, the game may work normally for a while, but then crashes again with the same error message, asking me to verify the integrity of game files. I have tried uninstalling and reinstalling the game, but the issue still persists.
My laptop(Dell G3 3579) specs are:
intel core i5- 8300h
nvidia  gtx1060
16 gb RAM

### Example Match ID (and possibly Timestamp)

7122766914-00:37(night)

### Screenshots

![error](https://user-images.githubusercontent.com/98306726/234169416-d6e5fd25-d854-4d5a-b748-57da760f6a15.jpg)

**Comment by cmq104** [comment]
same problem
![dota2crash](https://user-images.githubusercontent.com/68173644/234244381-8eb9919c-3f6a-45cc-9506-423a0cfc002d.jpg)
no such issue occurred before 7.33,crashed 2times after 4/21

**Comment by paulowak** [comment]
Meu jogo está apresentando este erro, minha placa de video nvidia gtx1060 pode ser por isto?

![image](https://user-images.githubusercontent.com/81195320/234400404-a121fbcc-f1cd-4419-bfa1-343e725d5dab.png)

**Comment by 1nuyasha** [comment]
> Meu jogo está apresentando este erro, minha placa de video nvidia gtx1060 pode ser por isto?
> 
> ![image](https://user-images.githubusercontent.com/81195320/234400404-a121fbcc-f1cd-4419-bfa1-343e725d5dab.png)

**Comment by ank-debug** [comment]
Any solutions ? facing same issue since 24th April.

**Comment by 1nuyasha** [comment]
> Any solutions ? facing same issue since 24th April.

no =.=

**Comment by Sir-Devil** [comment]
I've also had been having this same issue. I was wondering if it's a hardware issue since it's only on my new pc but it doesn't seem to be.

It doesn't goes away even after redownloading and running the integrity check multiple times

**Comment by ank-debug** [comment]
@Sir-Devil  Since when are you having this issue ? Mine started happening just a few days back. Before that I have been playing without any issues since last year.

**Comment by Sir-Devil** [comment]
@ank-debug I haven't played the game for about 2 months. Got back into it on the day 7.33 released and have been experiencing this ever since

**Comment by saw10** [comment]
Any solutions?

**Comment by Seam7** [comment]
Same issue happening to me

**Comment by yasseralzainy** [comment]
i have the same issue too

**Comment by enzok881** [comment]
does anyone find a solution? this happens after accepting a match or when you exit the game

**Comment by f1bb3r** [comment]
A couple seconds into a match, this exact same error appears. Game becomes barely playable, most of the sounds and some heroes' icons disappear and opening the main menu during the match completely crashes the game. Steam then forces me to verify integrity of game files and starts downloading 20+ GBs of files.

**Comment by hamidrogers** [comment]
This is just infuriating, Can't play like this , how many times does it need to verify the integrity? unplayable at this stage

**Comment by Sparkl1l** [comment]
i have the same issue too

**Comment by chihobix** [comment]
i have the same issue. can't fix it even after verifying integrity of game files

**Comment by chihobix** [comment]
> I have encountered this error while playing ranked matches. Is that the case for you all as well ? I am testing this on turbo matches. Played 5 matches today, didn't crash yet.

not only in ranked games, i recently played single draft low pri and i got the same error

**Comment by f1bb3r** [comment]
> I have encountered this error while playing ranked matches. Is that the case for you all as well ? I am testing this on turbo matches. Played 5 matches today, didn't crash yet.

Got this error in every turbo game I played, however, it started just recently, around 3 days ago.

**Comment by billyschmidt** [comment]
I'm continuing to have this issue, and the worst part is being punished for a single abandon (due to crash). I keep getting placed in the low priority pool and have to get 3 (now 4) wins to be able to play turbos again. This is pretty much to most frustrating thing ever, and steam support is useless/clueless.

<img width="1056" alt="Screenshot 2023-05-02 at 9 51 20 AM" src="https://user-images.githubusercontent.com/9101066/235732865-37845b44-14ee-4961-a524-51223c780de6.png">

How can I see my local crash logs?

**Comment by Guranis** [comment]
Hey guys i may have found something that might help some of u.

After the game crashing with this message i realized its never ever downloading anything.

So i decided to delete the file manualy and than do and Integrity check again.
HAve been playing the entire day long and had no issues what so ever ... so far ^^

Maybe u guys wanna give it a try and state if it worked for u

**Comment by f1bb3r** [comment]
> Hey guys i may have found something that might help some of u.
> 
> After the game crashing with this message i realized its never ever downloading anything.
> 
> So i decided to delete the file manualy and than do and Integrity check again. HAve been playing the entire day long and had no issues what so ever ... so far ^^
> 
> Maybe u guys wanna give it a try and state if it worked for u

Hm... good that you pointed that out! Thanks man, will try.

**Comment by a3v5** [comment]
> I'm continuing to have this issue, and the worst part is being punished for a single abandon (due to crash). I keep getting placed in the low priority pool and have to get 3 (now 4) wins to be able to play turbos again. This is pretty much to most frustrating thing ever, and steam support is useless/clueless.
> 
> <img alt="Screenshot 2023-05-02 at 9 51 20 AM" width="1056" src="https://user-images.githubusercontent.com/9101066/235732865-37845b44-14ee-4961-a524-51223c780de6.png">
> 
> How can I see my local crash logs?

dota crash log/dmp here:
steamapps\common\dota 2 beta\game\bin\win64
![image](https://user-images.githubusercontent.com/121069777/235780606-4260d25d-18f5-4989-a4bb-1078f124b38e.png)

also some log/dmp here:
Steam\dumps

![image](https://user-images.githubusercontent.com/121069777/235780430-30794b6a-d5f7-4a6c-8a79-a4d8bdcc265e.png)

**Comment by Sir-Devil** [comment]
> Hey guys i may have found something that might help some of u.
> 
> After the game crashing with this message i realized its never ever downloading anything.
> 
> So i decided to delete the file manualy and than do and Integrity check again. HAve been playing the entire day long and had no issues what so ever ... so far ^^
> 
> Maybe u guys wanna give it a try and state if it worked for u

For me, the error report is showing that "pak01.vpk" is corrupted but the problem is that there is no such file in the data folder

**Comment by ank-debug** [comment]
> Hey guys i may have found something that might help some of u.
> 
> After the game crashing with this message i realized its never ever downloading anything.
> 
> So i decided to delete the file manualy and than do and Integrity check again. HAve been playing the entire day long and had no issues what so ever ... so far ^^
> 
> Maybe u guys wanna give it a try and state if it worked for u

Which files did you delete exactly ?

**Comment by Guranis** [comment]
pak01_000
pak01_001

And then to be fair i tested my memory with mem test.
Seem like my memory wasnt working porper.
Some parts of it where not responding correctly.
Removed one of my 2 sticks (the faulty one)
And the game is running as im used to no crashes nothing.

So for me the problem is solved. Faulty memory.
Due to write and read error Dota was report a corrupt file and  never even downloaded something because the file itselfe was OK. My ram was screwing it up.

**Comment by chihobix** [comment]
> pak01_000 pak01_001
> 
> And then to be fair i tested my memory with mem test. Seem like my memory wasnt working porper. Some parts of it where not responding correctly. Removed one of my 2 sticks (the faulty one) And the game is running as im used to no crashes nothing.
> 
> So for me the problem is solved. Faulty memory. Due to write and read error Dota was report a corrupt file and never even downloaded something because the file itselfe was OK. My ram was screwing it up.

i tested my memory and deleted both these files and still get the same error. i played couple of single draft matches without a problem with most of them, but once i'm back on ranked i get the error on my first match and now i'm back to low pri

**Comment by billyschmidt** [comment]
Also still getting it after deleting the corrupt files, and in both turbos and single draft.

**Comment by Veloxigami** [comment]
I've been facing same error after latest patch.

**Comment by saw10** [comment]
One week and no solutions

**Comment by chihobix** [comment]
has anyone tried another steam account on the same pc?

**Comment by Flot12** [comment]
> т

yes, it doesn't help

**Comment by Flot12** [comment]
> Кто-нибудь пробовал использовать другой аккаунт Steam на том же компьютере?

yes it doesn't help

**Comment by fishybeanz** [comment]
> has anyone tried another steam account on the same pc?

Yes, just tried it, received same error from a different account( completely new).

**Comment by fishybeanz** [comment]
I've been going through the exact same issue now.

![image](https://user-images.githubusercontent.com/132766230/236631576-e0fe409c-30a2-49c5-862e-bde1ffbc3cf3.png)
![image](https://user-images.githubusercontent.com/132766230/236631612-ac74426b-2957-4d48-b267-3abd393002a9.png)
![image](https://user-images.githubusercontent.com/132766230/236631635-8783e399-d421-430b-b524-026aeafa09fd.png)
![image](https://user-images.githubusercontent.com/132766230/236631682-2456fd7c-f790-4391-8579-310139af6cc9.png)
GG Bois

**Comment by ank-debug** [comment]
I have been playing for a week now. Have not witnessed any crashes. Dont exactly know how it got resolved, I did reinstall my game once last week maybe it helped.

**Comment by chihobix** [comment]
i'm still having the same issue

**Comment by 1nuyasha** [comment]
Hey guys, it seems like my problem has been solved. I purchased two 8GB memory sticks to replace my original two 4GB sticks on April 3rd. I was able to play dota normally before the 7.33 update, and crashes only appeared after the update. I suspect that dota had higher requirements for memory stability after 7.33 update, and my new memory sticks may have had poorer compatibility and stability. (I bought cheap ones). Last week, I applied for after-sales service from the memory stick supplier and replaced them with two new memory sticks again. The compatibility and stability of the two new memory sticks are much better, and so far, I haven't encountered any game crashes. In conclusion, I think that the crashes were caused by memory failures due to the poor stability of the cheap memory, and replacing the old memory sticks with newer ones that have better stability might solve this problem. Perhaps you guys can give it a try. =.=

**Comment by Cyproximity** [comment]
I'm also encountering this issue since patch 7.33b. I tried to re-install the game and it won't fix the problem. btw, my system specs are AMD 3600x CPU and 16 GB of RAM. 

the scenario when it's happening. when there's a lag spike happens there's a small chance that the popup error will show and the game itself will crash. and then when I open Dota 2 again it will verify the game files. 

not sure if this is a memory issue. because when I play other games, not Dota. my system is totally normal and I never encounter any BSOD. I believe  this is Dota 2 issue

**Comment by YuriJr** [comment]
Just wait valve resolve it, i think that’s not specs problem

**Comment by hamidrogers** [comment]
Hey guys I had this problem for at least 2 weeks, 2 days ago I completely deleted steam I mean shift deleted everything, Then installed a fresh one on another Hard drive , And downloaded the game again, past 48 hours not a single crash, Try installing it on another hard drive, this fixed it for me

**Comment by Teikoboii** [comment]
I removed PUBGs AC-Tool Xhunter1.sys. No issues since.
This also cause some memory integrity issues in win11 itself.

```
net stop xhunter1 
reg delete HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\xhunter1 
del C:\Windows\xhunter1.sys
```

**Comment by celestynek** [comment]
Unsubscribe every map from dota , it works

**Comment by duranddsouza** [comment]
First of all so happy I found this thread. Be hounding my desktop manufacture for a week.

7950 AMD, 7900 XTX and  have the issue. So clearly not spec related

Reinstalled for steam and dota2 fixes it for a couple days (4 or 5 games) and its back again (Followed this step twice)

Valve should atleast acknowledge the issue, can someone with a decent twitter follower count please tweet about this only way will get a response

**Comment by victorcossioq** [comment]
@celestynek I did that but the issue has happened again today :(

**Comment by jaredeg** [comment]
Also having this (got a 5900X 4090)... could it be a Rebar issue?

**Comment by WuQingWen** [comment]
Hey guys，I fixed the problem. This is a memory compatibility problem. My original memory was 16G x 2, XMP 3666 MHz, but now it has been adjusted to 3600 MHz, The problem did not appear again.     @1nuyasha

**Comment by 1nuyasha** [comment]
> 
yeah, I also believe that incompatibility of the memory is the main culprit. My game hasn't crashed for three weeks ever since I replace the old RAM sticks with new ones.

**Comment by Sir-Devil** [comment]
I don't know if the memory is the only reason since I had the same issue but the memtest showed no errors. So, I tried what Guranis mentioned above and deleted the two files and it seems to have fixed the problem but have to play more to be sure.

**Comment by hamidrogers** [comment]
I Shift-deleted The whole game, then reinstalled in on another hard drive, fixed it for me ive been playing without any problems for 2 weeks, Didnt change my RAM or anything, Just installed it on another Drive, hope it helps coz i know how annoying this crash is

**Comment by MohammadHk85** [comment]
any soloutions? facing the same problem these days.

**Comment by michaellongge163** [comment]
same error

**Comment by kerbyc07** [comment]
> pak01_000 pak01_001
> 
> And then to be fair i tested my memory with mem test. Seem like my memory wasnt working porper. Some parts of it where not responding correctly. Removed one of my 2 sticks (the faulty one) And the game is running as im used to no crashes nothing.
> 
> So for me the problem is solved. Faulty memory. Due to write and read error Dota was report a corrupt file and never even downloaded something because the file itselfe was OK. My ram was screwing it up.

HELLO, got the same problem and the solution i had was to rest my pc for like a day, it happened to me at night and used it again the next night, then i deleted the two files. it downloaded something and validated the whole 47gb file and i had a quick verification of my file and then i got to play 5 games without it happening to me.

**Comment by victor-pogor** [comment]
Same here. Tried everything but nothing helps

![image](https://github.com/ValveSoftware/Dota2-Gameplay/assets/24962085/5d026dce-2694-4190-99dc-df3423435ad9)

**Comment by rolandabb** [comment]
Guys after lots and lots of attempts, I finally found a working solution.
So after trying to manually move the files from local disk D to local disk C, 2 files : pak01-325 and pak01-326 weren't copying at all. So I manually deleted them, and then verified integrity of game cash, which got them to be redownloaded, and then it worked perfectly fine.
So you just need to find which pak file is corrupt, delete it and then verify integrity of game files.

**Comment by Nerka42** [comment]
i know this only for dota2 but i am having the same error whit csgo2 and i dont see other people tallking about it

**Comment by Astralopitek146** [comment]
> 
To find whick pak is corrupted should I just copy these paks from one place to another? And if pak doesn't copy it will means that it is corrupted, am I right?

**Comment by Astralopitek146** [comment]
> Guys after lots and lots of attempts, I finally found a working solution. So after trying to manually move the files from local disk D to local disk C, 2 files : pak01-325 and pak01-326 weren't copying at all. So I manually deleted them, and then verified integrity of game cash, which got them to be redownloaded, and then it worked perfectly fine. So you just need to find which pak file is corrupt, delete it and then verify integrity of game files.
 

To find whick pak is corrupted should I just copy these paks from one place to another? And if pak doesn't copy it will means that it is corrupted, am I right?

**Comment by rolandabb** [comment]
> > Guys after lots and lots of attempts, I finally found a working solution. So after trying to manually move the files from local disk D to local disk C, 2 files : pak01-325 and pak01-326 weren't copying at all. So I manually deleted them, and then verified integrity of game cash, which got them to be redownloaded, and then it worked perfectly fine. So you just need to find which pak file is corrupt, delete it and then verify integrity of game files.
> 
> To find whick pak is corrupted should I just copy these paks from one place to another? And if pak doesn't copy it will means that it is corrupted, am I right?

Yes, but make sure to keep copying so that you make sure no other pak file is also corrupt.

**Comment by jaredeg** [comment]
> > Guys after lots and lots of attempts, I finally found a working solution. So after trying to manually move the files from local disk D to local disk C, 2 files : pak01-325 and pak01-326 weren't copying at all. So I manually deleted them, and then verified integrity of game cash, which got them to be redownloaded, and then it worked perfectly fine. So you just need to find which pak file is corrupt, delete it and then verify integrity of game files.
> 
> To find whick pak is corrupted should I just copy these paks from one place to another? And if pak doesn't copy it will means that it is corrupted, am I right?

I manually deleted the game and still run into the issue (although a bit more infrequent now)

**Comment by Capt1nSweatPants** [comment]
I have been agonizing over this problem for 2 weeks. I did all the software fixes but failed, then I tried reseating (disconnecting and reconnecting) my Hard Drive and It got fixed. Maybe this might help some of y'all.

**Comment by yems730** [comment]
Just got the error today, tried reinstalling,integrity clear download cache, now manually deleting the dota folder and then i'll try again.

**Comment by CloudDeath** [comment]
I have been trying to deal with this error for a month, tried all possible options offered on the Internet. 
BUT! The problem turned out to be in my SSD. When I installed the game on HDD - everything was fine, there were no errors. But when I installed dota on SSD (for faster loading and increased fps), not only it was not always possible to install it because of errors in steam itself (disk read error, etc.), but also constantly appeared this error.

My conclusion of this "investigation" is as follows: check the state of your hardware :) (drive, RAM, etc.).

The error is clearly not in the code of the game.

Now I have bought a new SSD, I do not see any errors at all.

**Comment by PasswordCodmNN** [comment]
https://github.com/ValveSoftware/Dota2-Gameplay/issues/8584#issuecomment-1566447013         hey bro

**Comment by yems730** [comment]
Same error for me both for dota and csgo2. Checked ram,cpu,swapped psu checkd motherboard no errors. All after their latest update.

**Comment by alexsanderar-nexstar** [comment]
Does anyone got the fix for this frustrading issue?
I'm still getting this error at least twice day. 
![Screenshot from 2023-11-05 21-13-51](https://github.com/ValveSoftware/Dota2-Gameplay/assets/141274284/232c9e09-eba7-4461-a950-9bd930976248)

**Comment by jaredeg** [comment]
Does everyone running into this issue have a specific type of SSD? I am running Dota 2 on a  Samsung 1 TB 980 Pro

**Comment by Juks16** [comment]
I have the same issue, in my both PC and laptop. it doesn't matter what specs of your computer, I think the problem is in the new patch of dota2 or some kind of hacker messing our files. in my case it occurs a lot in ranked mode, buying new ram, verify game integrity, redownload game will not work.
PC (Ryzen 5 3500x/3200mhz 16Gb RAM/MSI B450/MSI Rx 570/ M.2 512 GB)
Laptop (Aspire 572G/8Gb RAM)

for a temporary a solution I recommend to install steam & dota2 in external drive/ use deep freeze or similar app/have a better AV.

**Comment by lucasskora** [comment]
Solving the problem, as it was something related to related files, led us to believe that it was something about the SSD.
What I did was, adjust the cables connected to the SSD and put a new one in the one where Dota was installed. In fact I uninstalled and installed again after this process.
I've been playing now for over an hour, I watched calm matches, before it was 10 minutes and it was already crashing.
Now let's see if we will keep faith.

**Comment by bkaankose** [comment]
I can confirm this issue happens time to time. Started this week for me. Sometimes I got stuck at %0 verify, sometimes my verification continued.

My steam installation and games are in different disks. Both are different SSDs, not only partition. Maybe this may help the devs.

**Comment by 50centaur** [comment]
This seemed to work for a little bit but then I got the error again a few
days later. For the downloading content error, I found a workaround by
logging into a different steam user account and then downloading from
there, and then re-logging back in once the download is complete.

**Comment by Recall703** [comment]
I faced this bugs just now 
and my game was just started:

My PC specs are:
AMD Ryzen 3700X
nvidia RTX 2070S
32 GB RAM

I have tried the following:
1. Verify integrity files -> Stuck at 0% for 5 mins
2. Clear download Cache, Verify integrity files -> Stuck at 0% for 5 mins
3. Restart Steam, clear download cache and clear download Cache, Verify integrity files -> Stuck at 0% for 5 mins
4. Restart PC

All those doesn't work for me
end up I need to reinstall the entire game in order to play
and thanks to that I faced a penalty 

when I talked to my friends (who live in the same house, and using the same network)
we found out that was a small update (26.75KB)

I not sure this may help or not, and I sincerely hope it will get fixed ASAP, as this crash happened a few times, and all my solution was to reinstall dota 2

**Comment by jaredeg** [comment]
Tried many of the post solutions in this thread, and they all seem to work, and then the crashes would come back.

The only thing that actually resolved my issue was  I start doing memtest86 on my memory, and I ended up finding out that I had unstable memory at XMP profiles. Down clocking the memory, resolved my issue for good.

**Comment by andkalinin** [comment]
guys, if you check the integrity of the files and nothing happens.I run the game through a file.
For example C:\user\Steam\steamapps\common\dota 2 beta\game\bin\win64\Dota.exe.
And it works. I don't know how

**Comment by 50centaur** [comment]
So I've managed to recreate this error consistently by Connecting to a
NordVPN server, Disconnecting, launching dota 2.

**Comment by yihleego** [comment]
So Valve still hasn’t fixed this bug 🤮.

**Comment by Never2land** [comment]
I got banned for doing nothing, it's not my fault the game keeps crashing. Tried everything in this post, but nothing worked. Crashing everything other days makes this game unplayable. Especially it happens right when a new game starts and it won't allow me to re-download/verify game content..

**Comment by victorcossioq** [comment]
I solved this by buying a new dram stick to replace the oldest one and removing and reputting all the ram sticks. You can try it.

**Comment by Never2land** [comment]
> I solved this by buying a new dram stick to replace the oldest one and removing and reputting all the ram sticks. You can try it.

Thanks for the advice, will try it.

**Comment by Phara0h** [comment]
Brand new computer with brand new SSD and its happening to me

**Comment by victorcossioq** [comment]
> Brand new computer with brand new SSD and its happening to me

Remove and reput your RAM sticks. that fixed my pc

**Comment by grandleemon** [comment]
I just started with an .exe file in the dota folder, worked for me

**Comment by Phara0h** [comment]
> > Brand new computer with brand new SSD and its happening to me
> 
> Remove and reput your RAM sticks. that fixed my pc

I got it working but turning off xmp for my ram but kinda sucks since it drops it 2ghz (from 8ghz to 6) without it

**Comment by AaronAcerboni** [comment]
Same, getting the above error. I tried re-installing dota2 but no dice. Also verifying the integrity of files via the steam client just leaves the progress stuck at 0%.

:(

**Comment by Never2land** [comment]
Turned off xmp, reinstalled ram, was good for a while. Got my First crash today.... Feels like this update breaks something again. Or I need to reinstall ram again...

**Comment by venom-mancer** [comment]
> Turned off xmp, reinstalled ram, was good for a while. Got my First crash today.... Feels like this update breaks something again. Or I need to reinstall ram again...

try to install the game on another drive, for example I had a ssd and the game was crashing after playing 1 or 2 games , then I installed the game on another ssd I have. it works perfectly and never crashed.

**Comment by LiangNex** [comment]
So far the only solution is waiting for about 10 ~ 20 mins if you don't want to reinstall the game..
The game will magically start to verify itself as long as you wait for that long time, then everything works as expected.
And for sure, your current battle is abandoned and you will get several reports, WTF

Guess there might be something buggy with the file system of Dota2.🥶

**Comment by LiangNex** [comment]
Plus one more thing, the most possible reason of this bug is insuffcient runtime memory or something like that.
I only suffer this bug while I am playing dota2 with mulitple mobile simulators(like three or more) runing... and it happened twice for me in this case...

**Comment by GabeDia** [comment]
Sure Valve bush this off as faulty RAM even though ONLY dota is exposing such an issue on our systems. But your crappy devs should be held accountably for the bugged behavior of the steam client afterwards which is the real cause of the abandon and bans we are receiving.

After crashing dota2 will be stuck on either verifying or downloading at 0% with no way to recover before an abandon. I've tried keeping a copy of the vpk's to copy manually when it's corrupted and crashes. Even after copying good vpk's it's stuck at 0% despite cleaning steam download cache, restarting steam and restarting the PC. Some 30 minutes later magically the validate/download works and finishes, leaving no chance for me to re-join queued dota2 game and receiving a ban.\

So why is Steam preventing us from reconnecting and forcing an abandon and ban on us? Fix this shit.

**Comment by madroot** [comment]
I have exactly the same problem! VALVE FIX IT!!!

Has anyone found a way to at least pass the file check quickly and have time to log into the game?

**Comment by madroot** [comment]
valve pay attention to this problem!!!!!!!!!!!!!!!!!!!!

**Comment by UrielMy** [comment]
Same problem, already get 20 low priority games, gg wp

**Comment by Bibovsky** [comment]
got it first time, now im in a club too, valve fix 0% integrity verification atleast

**Comment by mikalai-dzemidzenka** [comment]
Same problem

**Comment by James-Buckland** [comment]
Me too :(
Is this an actual bug tracker or just an echo chamber for people to cry out in?

**Comment by 1nuyasha** [comment]
> Me too :( Is this an actual bug tracker or just an echo chamber for people to cry out in?

Actually a bug tracker, but valve do not pay attention to this issue at all [Sadge].
@jeffhill @BeeeQueue @danieljennings @wykrhm

**Comment by madroot** [comment]
Now I have again received 5 games of low priority due to this bug, valv completely does not care about this problem

**Comment by GabeDia** [comment]
Cool another abandon. At this point I feel like Valve is doing this intentionally by adding some timeout on the cloud side which won't let us repair our local installations. 

I've tried local game restore feature, clearing steam download cache, rebooting, changing download server region and even manually copying backups of the vpk - still dota is stuck from launching or doing anything until it just happens to start validating some 30 minutes later.

**Comment by Never2land** [comment]
Another day, another abandon....

**Comment by funatomana** [comment]
I also encountered the same problem
cpu:amd 7800x3d
gpu:rtx 4070super

**Comment by SolidSnake4** [comment]
> pak01_000 pak01_001
> 
> And then to be fair i tested my memory with mem test. Seem like my memory wasnt working porper. Some parts of it where not responding correctly. Removed one of my 2 sticks (the faulty one) And the game is running as im used to no crashes nothing.
> 
> So for me the problem is solved. Faulty memory. Due to write and read error Dota was report a corrupt file and never even downloaded something because the file itselfe was OK. My ram was screwing it up.

I can confirm It's memory problem, I bought new rams yesterday and everything is working fine without any problems. 
Thank yo so much.

**Comment by GabeDia** [comment]
If dota is causing memory issues that only a heavy memory stress test can cause then it's honestly just bad programming by Valve, I like many others have an otherwise stable system.  Probably it's embarassing for them to acknowledge this issue nevermind add a workaround for users because of their poor design.

It's simple what Valve should do:

-Add a way to check for the corrupted state that won't risk an abandon so it can be fixed before queuing - sometimes it occurs on splash screen during launch, demo'ing hero, entering hero selection or entering game from hero selection.

-Allow users to repair/validate the game at any time - stop locking people's game via steam cloud when game gets corrupted either entering or already in-game, forcing abandons.  What is one good reason to force an abandon instead of allowing the option of repair/rejoin?

This isn't even specific to dota, a quick search online shows that counter-strike also uses the .vpk filesystem and users have the same issues.

**Comment by anko** [comment]
@GabeDia 

> If dota is causing memory issues that only a heavy memory stress test can cause then it's honestly just bad programming by Valve

I think this is a misunderstanding.

Software is written with the assumption that the hardware it runs on is working correctly, and that it will work correctly regardless of system load. Memory malfunctions cause basically [undefined behaviour](https://en.wikipedia.org/wiki/Undefined_behavior): It violates all kinds of fundamental assumptions about how computers are supposed to work.

It is usually better for software to crash when it detects corruption and show an error message, rather than to try to continue with corrupted data, since something worse might happen if it did continue, like overwriting your settings with random values or something. With random memory corruption, pretty much anything can happen, including the program's running code being unpredictably changed, because that too is loaded into memory. Many uncontrolled faults will also cause the OS to immediately stop the program, with no chance to display an error that might help diagnose it.

If your system is unstable under high load, chances are it's a little bit unstable under low load too, since whatever memory access pattern causes it to malfunction will sometimes randomly happen even at low load. You may see random crashes or freezes. So if the problem is memory, it would really be best to address this issue at the root.

- - -

As as I can tell, multiple people in this thread who managed to solve this problem had RAM hardware issues.

I would therefore recommend anybody seeing this issue to run a [memtest](https://memtest.org/), to check if the issue is memory hardware. A full memtest takes a couple of hours, and can be left unattended.

If the memtest fails, consider these possibilities:

- If you built your PC yourself, and are using the motherboard's built-in XMP settings, those might be bad.

  For example, I had to overvolt my RAM by a small amount over the built-in XMP settings, to get it stable at higher frequencies. This was even though my RAM is on the motherboard's officially supported hardware list! Again, you can confirm stability with memtest after tweaking settings. (I did not have this specific issue, but I had other crashes that disappeared when I did this.)

- If you have multiple RAM sticks, it might be that only one is bad. Try removing one and rerun memtest. This might save you money buying replacements.

- If you are on Linux, and memtest only finds a few memory regions that are problematic, consider [these workarounds to disable bad memory regions on boot](https://unix.stackexchange.com/questions/75059/how-to-blacklist-a-correct-bad-ram-sector-according-to-memtest86-error-indicati).

- If you are on a laptop: Some laptops allow RAM to be replaced easily, by opening the back cover. Check repair guides for your model. But some manufacturers solder RAM chips in, meaning you need specialist tools to replace them. Ask a PC repair shop.

If memtest comes back clean, and you _still_ see this issue after verifying game files, then it could be an actual Dota bug. If that is the case for you, please comment for completeness!

**Comment by GabeDia** [comment]
> > If dota is causing memory issues that only a heavy memory stress test can cause then it's honestly just bad programming by Valve
> 
> I think this is a misunderstanding.

It's really not - like I said many of us have a system that is completely stable aside from this issue with DoTA. If DoTA's programming is exposing issues that only a dedicated memory stress test would, how is that a good software design?

Secondly, sure your post is okay general advice about RAM instability - but that's not the point, valve is actively punishing users who are affected by this issue instead of coming up with any way to prevent ruined games in the first place. If you're okay with that then don't bother posting in this thread.

**Comment by anko** [comment]
> If DoTA's programming is exposing issues that only a dedicated memory stress test would, how is that a good software design?

Personally, I'm happy that Dota makes maximum use of the hardware that I paid for. If it didn't, I would be annoyed, and I would consider _that_ a bug.

Either way, I hope this gets fixed so everyone could play without crashes. I don't work at Valve, so I don't know if a software fix is possible, and other commenters seem to have successfully fixed this issue at the hardware level. That's why I focused on hardware-related advice. Just trying to help where I can. If you or anybody else with this issue needs help checking if it's a hardware problem, let me know and I'll try to assist.

**Comment by adambotley** [comment]
> If memtest comes back clean, and you still see this issue after verifying game files, then it could be an actual Dota bug. If that is the case for you, please comment for completeness!

This is the case for me. Ran several memtest cycles to try and spot memory issues, but I'm not seeing any errors. Also getting crashes with XMP disabled completely to verify that was not the cause either.

**Comment by GabeDia** [comment]
@adambotley if you want to try another testcase, testmem5 is another good program and should require less cycles than memtest. Per anko's comment you can try tinkering with bios settings for ram timings, voltages, etc. to increase stability - this may have reduced occurence for me but not eliminated it. 

If disabling XMP settings for RAM didn't help then probably further tuning won't do much, also you're losing a lot of performance (on games besdies dota) especially if its a newer system.

The next less-obvious cause would be a failing hard-drive, though I think the majority of these cases are to do with RAM. You can check overall state of the drive with CrystalDiskInfo utility.

To make my other point one more time - if a user is disconnected from their internet loading into, selecting hero or mid-game they can re-join. In the case of this crash the steam cloud actively prevents repair of the game (i.e. stuck at 0% download/verify) forcing an abandon, leaving no option to re-join in time.

**Comment by felix-ht** [comment]
This seems like its a ram issue. This also means that files never get corrupted. After i get the issue i cannot resart dota though stem as steam it stuck downloading.

For me i clould just go into the dota game folder and directly run the *.exe - i could play the game afterwards. After some time the steam client also recovered and did a download to "fix" the game.

**Comment by astashys** [comment]
Resetting the values of the AMD PBO2 Curve optimizer resolved the issue in my case.
Interestingly, I have tested the values via different tools like OCCT, and y-cruncher and they passed but Dota crashes when starting a match in multiplayer.

However, I want to tune the Curve optimizer with values that would not crash. Testing values in a real match is not profitable since there is a 100% chance of getting low-priority penalties or if you try to reproduce with bots, it rarely reproduces and you still need to wait 10-20 minutes until Steam comes to his senses.

Has anybody found a test that could reproduce the issue without any penalties? Maybe some synthetic test or playback of load?

**Comment by GabeDia** [comment]
![valve_sucks](https://github.com/ValveSoftware/Dota2-Gameplay/assets/5883430/f39271e6-eedd-4ef0-922e-fbb59cf55f4d)

Machine was stable for a month after tweaking timings then crashed mid-game. I was able to launch via the the dota2 executable directly but then got a VAC message in-game and wasn't able to reconnect, resulting in abandon. 

So the Valve team is not skillled enough or don't care to differentiate between this crash and cheats (or otherwise odd client behavior), hence the the forced abandons. My guess is the vpk files themselves aren't corrupted, only the memory is once loaded into RAM - so the file scan is sneaky way of checking for cheats.

**Comment by olishiz** [comment]
![image](https://github.com/ValveSoftware/Dota2-Gameplay/assets/22555339/d08fbf26-bef8-4811-b786-25ba3118d63c)

I am having this issue every 2-3 days after starting and playing my Dota2. This has been frustrating for me and I hope Valve is able to fix this issue. Everytime, this will lead me to abandon my game and I got into penalty (for which I did not sign up to).

Does changing the location of the files to other drive seems to work? How can I permanently solve this issue?

**Comment by TomCockram** [comment]
This issue is still happening.

**Comment by Dode123** [comment]
Still happening here as well.... Every now and then. Never get this issue in other games.

**Comment by andriandurlestean** [comment]
I can't even download Dota, check the files 0% integrity, freeze, or delete everything, even Steam, cant download dota :(

**Comment by CodiciaDotaYT** [comment]
para todos los que tienen ese problema, al parecer es un conflicto con el hardware, para ser más específico, la memoria ram, intenten cambiar la frecuencia de las memorias ram de su pc en la configuración avanzada de la bios, por ejemplo de 3600hz --> 3200 , prueben cambiar a diferentes frecuencias hasta que su problema se solucione, agradecería respuestas para confirmar que los he ayudado

**Comment by GabeDia** [comment]
I lowered my RAM timings and it seemed to fix it for a while - but yet again here I am. Just reiterating how garbage it is that a crash triggers VAC, holding game validation at 0% to force an abandon which ruins the game for others as well.

**Comment by pocobio** [comment]
This shit moved to cs2, FUCK valve lmao

**Comment by deyvtech** [comment]
It happened again

**Comment by felix-ht** [comment]
Bought a different ram kit - with slightly lower base clock. And listed in the compatibility sheet of the mainboard. Seems to have fixed it.

**Comment by andriandurlestean** [comment]
worst part, i cant verify or redownload dota, command "steam://flushconfig" helped to renstall the dota

**Comment by andriandurlestean** [comment]
Well did memtest86, one of my ram sticks died, I think this is the problem
![Screenshot_20241018-213854](https://github.com/user-attachments/assets/8dfb6167-0604-4776-8465-c70d6bfa813e)

**Comment by olishiz** [comment]
Hey guys, so I've concluded that when I changed my RAM sticks - I found out that 2 of my 16gb sticks are faulty after being used for 4+ years? 

# Issue
- So I found out issue would be faulty rams.

# Problems Fixed
- Try changing to new RAM sticks and the issue would be resolved. I change since last 4 months ago - and I have not have this issue anymore. It is not recurring anymore.

Thanks guys.

**Comment by leobrch** [comment]
Hi,

I somehow managed to fixed it by using these launch options
![image](https://github.com/user-attachments/assets/969df539-4838-4770-b079-a3656f16c70c)

`-dx11 -novr -novid -safe_mode`

I'm on linux and I hope it helps
![image](https://github.com/user-attachments/assets/9c56d221-42d0-4835-8b28-e3e936edbcc1)

**Comment by ThatsDrew** [comment]
I am also having the exact same issue. RAM was already tested. RAM XMP was disabled. CPU overclock was disabled. Game was reinstalled, tried the launch options mentioned in the above comment . Nothing works. None of the other games in my steam library (including other online multiplayer games with anticheat) have any issues running. If anyone could please take a look at the issue that would be great. Please let us know which kind of logs you need or how we can be further of help. This is extremely frustrating :(

see:
#22472 
#20645 
#20478 
#19695 
#15737 
#16483

**_might be worth merging these/marking them as duplicate and closing them ;)_**

**Comment by lewisdiamond** [comment]
Seeing .pak file corruptions as well, so far it only happened during pick phase and during the transition between pick phase and the actual game. My RAM is fine according to memtest.

**Comment by andriandurlestean** [comment]
After buying new ram cuz was corrupted, no more issue

**Comment by duranddsouza** [comment]
How did you confirm your ram was corrupted ? What test did you run. I had a
brand new setup last year all test would pass but would still have the
issue.

**Comment by andriandurlestean** [comment]
I did memtest86, you can see my screenshot above, bought new ram, tested that one, no errors, no more Dota crushes

One of my friend had no errors but to much ram OC, he downgraded the ram OC and no more crushes.

**Comment by lewisdiamond** [comment]
Memtest reports no problem for me, still this happens and puts me in LP.

**Comment by felix-ht** [comment]
also got RAM at slower clockspeed and now the problem is gone.

**Comment by qherreros** [comment]
It has been more than a year now that CS2 and DOTA have weird bugs linked to RAM. From what I've read, most people had 0 problem with their computer, only CS2/DOTA are crashing. Replacing your RAM or downclocking it to solve a specific game issue sounds like an horrible option.

**Comment by andriandurlestean** [comment]
Lol after changing ram having all green in memtest, no OC, no errors, only valve game crushes, why can't verification content be instant, waiting for 5 min to be low priority, beta game will remain beta

**Comment by GabeDia** [comment]
Yeah this is exactly my point I was trying to make @andriandurlestean there's a fundamental issue with steam's pk3 files. I suspect steam made related changes to this, as now my game seems to crash on launch though not mid-game/character selection anymore.

**Comment by ziggi** [comment]
Got same issue on a new PC, memtest is OK

![Image](https://github.com/user-attachments/assets/04c6108c-ef62-4a1f-b1fb-8d8211973708)

![Image](https://github.com/user-attachments/assets/9680960f-ad25-4c35-a85d-a0de5caecbf9)

![Image](https://github.com/user-attachments/assets/2ccd6190-4fa3-458e-9aa9-90357940e5cf)

**Comment by duranddsouza** [comment]
I had 4 * 16 ram and refused to remove them for 1 year I am in this chain
as how can my brand new ram be the problem.

Removed two stick and 6 games no problem so far. It’s the ram guys

**Comment by qherreros** [comment]
I've been able to solve this problem on linux adding LD_PRELOAD="". Solved all issues I had. It was not the RAM, obviously, as only 2 games were crashing.

**Comment by GabeDia-Sav** [comment]
Can valve at least enable the re-verify of game files when it crashes? It just happened to me right on game load - this is so obnoxious that it blocks game validating and forces abandon on our accounts.

**Comment by GabeDia** [comment]
So when will valve chage it so we can re-validate after a crash during a game? Just got another abandon, crashed right after hero selection which ruins behavior score.

**Comment by taraih** [comment]
Happens a lot more now since the latest 7.38 patch. In thousands of games I almost never had any abandon. Now 2 abandonds in 3 days. The issue only exists with Dota. Can you fix your game already? If the files would verify instantly but it waits 15-30 min so the game is done and your account gets ruined. Its on YOUR END valve. NO other game has those issues. My pc runs WELL. What a joke. Literally the WORST game you could have this problem in with abandon system.

**Comment by RetardedDog** [comment]
Same. 8100 hours without a single issue, and now I am constantly getting this error after the 7.38 happened. My PC ain't in the best condition, that is true. But considering that even getting a new RAM is only a temporary solution, and that some people are fixing it by doing some completely obscure and random things, and that the only one 100% fix is to build a new PC from the scratch, idk if it's worth it to try and fix this issue. 
I'd rather just take it as a sign and go touch some grass. Thanks Valve, I guess...

**Comment by OldManFromWest** [comment]
I think that I found the solution. Short guide in here (you can use translators):
https://steamcommunity.com/sharedfiles/filedetails/?id=3511740782

**Comment by Frodolele** [comment]
I found solution for gigabyte b650m, i think this work on all gigabyte motherboard:
1. If you have memory have 6000Mhz
2. If you have XMP profile
3. In bios u need activate 2 options: low latency support, handwith

**Comment by scintilla083** [comment]
### **Solution**

**Summary**

Many users have been experiencing a recurring crash in Dota 2 (Source 2 engine) for the past few years.
After analyzing the crash using WinDbg, it appears the crash is not caused by an access violation or invalid memory operation, but by a deliberate call to RaiseException() inside the Source 2 runtime — specifically inside tier0!Tier0GenericMiniDumpHandler.


**Key Point**

The game loads tier0.dll from
`D:\SteamLibrary\steamapps\common\dota 2 beta\game\bin\win64\`
but Steam injects its own tier0_s64.dll from
`C:\Program Files (x86)\Steam\steamclient64.dll.`

This is precisely why some people completely lost the error when running without Steam integration.
like this - `dota2.exe -nosteam`

Solution **(Step-by-Step)**
1)Close Steam and Dota 2
2)Locate conflicting DLLs 
    D:\SteamLibrary\steamapps\common\dota 2 beta\game\bin\win64\
    Look for tier0.dll and other DLLs like vstdlib.dll, tier1.dll
3)Copy these DLLs to a safe folder, e.g., backup_DLLs
4)Remove the conflicting DLLs
`tier0.dll → tier0.dll.bak
vstdlib.dll → vstdlib.dll.bak
tier1.dll → tier1.dll.bak
`
5)Verify game files via Steam
6)Launch the game


P.S.
You can try this bash script and not delete any dlls or do anything else.
This should preload the necessary DLL and help avoid file conflicts.
```
@echo off

setlocal

set "PATH=D:\SteamLibrary\steamapps\common\dota 2 beta\game\bin\win64;%PATH%"

start "" "steam://rungameid/570"

timeout /t 3 /nobreak >nul

endlocal
exit /b 0
```
Script result (load main dll)- 

<img width="1507" height="625" alt="Image" src="https://github.com/user-attachments/assets/e895be46-540c-4a72-88f2-a16083911830" />
